compiler/rustc_data_structures/src/vec_cache.rs

```diff
@@ -68,22 +68,13 @@ impl SlotIndex {
     // slots (see `slot_index_exhaustive` in tests).
     #[inline]
     const fn from_index(idx: u32) -> Self {
-        let mut bucket = match idx.checked_ilog2() {
-            Some(x) => x as usize,
-            None => 0,
-        };
-        let entries;
-        let running_sum;
-        if bucket <= 11 {
-            entries = 1 << 12;
-            running_sum = 0;
-            bucket = 0;
-        } else {
-            entries = 1 << bucket;
-            running_sum = entries;
-            bucket = bucket - 11;
+        if idx < 4096 {
```

Comment:
Can these magic numbers be turned into constants? I see the old code had `1 << 12` hardwired. While you're in here...

Reply:
Done.